graylog2-server/src/main/java/org/graylog2/streams/StreamRouter.java

```diff
@@ -39,24 +42,33 @@
 public class StreamRouter {
     private static final Logger LOG = LoggerFactory.getLogger(StreamRouter.class);
 
-    private static final long ENGINE_UPDATE_INTERVAL = 1L;
-
     protected final StreamService streamService;
     private final ServerStatus serverStatus;
+    private final ScheduledExecutorService scheduler;
 
     private final AtomicReference<StreamRouterEngine> routerEngine = new AtomicReference<>(null);
+    private final StreamRouterEngineUpdater engineUpdater;
 
     @Inject
     public StreamRouter(StreamService streamService,
                         ServerStatus serverStatus,
                         StreamRouterEngine.Factory routerEngineFactory,
+                        EventBus serverEventBus,
                         @Named("daemonScheduler") ScheduledExecutorService scheduler) {
         this.streamService = streamService;
         this.serverStatus = serverStatus;
+        this.scheduler = scheduler;
+
+        this.engineUpdater = new StreamRouterEngineUpdater(routerEngine, routerEngineFactory, streamService, executorService());
+        this.routerEngine.set(engineUpdater.getNewEngine());
+
+        // TODO: This class needs lifecycle management to avoid leaking objects in the EventBus
+        serverEventBus.register(this);
+    }
 
-        final StreamRouterEngineUpdater streamRouterEngineUpdater = new StreamRouterEngineUpdater(routerEngine, routerEngineFactory, streamService, executorService());
-        this.routerEngine.set(streamRouterEngineUpdater.getNewEngine());
-        scheduler.scheduleAtFixedRate(streamRouterEngineUpdater, 0, ENGINE_UPDATE_INTERVAL, TimeUnit.SECONDS);
+    @Subscribe
+    public void handleStreamsUpdate(StreamsChangedEvent event) {
```

Comment:
Adding `@SuppressWarnings("unused")` would disable the compiler warnings (and make the IDE happy).